batch/batch/driver/instance_collection/pool.py

```diff
@@ -380,22 +413,40 @@ async def create_instances(self):
 
         AUTOSCALER_LOOP_RUNS.labels(pool_name=self.name).inc()
 
+        for metric in AUTOSCALER_HEAD_JOB_QUEUE_READY_CORES._metrics.values():
+            metric.set(0)
+        for metric in AUTOSCALER_HEAD_JOB_QUEUE_N_INSTANCES._metrics.values():
+            metric.set(0)
```

Comment:
Why explicitly set 0 here? If we get scrapped during between here and after the following await we'll get a metric value of 0 which is wrong. I think we just need to set the final answer.

Reply:
I'm having trouble with this. The issue I was seeing was the old values for regions that were no longer requested and not part of the head queue would be stuck on the old metric forever. I was trying to fix that, but it wasn't working. Can you please help me figure out the correct way to do this?

Reply:
Ah, I understand now why you were doing it this way. I think it would be more readable if instead of the below iteration over `items` do something like the following:

```python
for region in app['regions']:
    requested_cores_mcpu = head_job_queue_region_str_ready_cores_mcpu.get(region, 0)
    AUTOSCALER_HEAD_JOB_QUEUE_READY_CORES.labels(pool_name=self.name, region=region_str).set(requested_cores_mcpu / 1000)
```

Reply:
The issue is that each item in the job queue has a region string that can have multiple regions. If you do it this way, you'd need all possible combinations of regions.

Example:

`['us-central1']`
`['us-central1', 'us-east1']`
`['us-central1', 'us-west1']`